app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    def inner(*args, **kwargs) -> Any:
+        if func in cached_results and args in cached_results[func].keys():
+            print("Getting from cache")
+            return cached_results[func][args]
+        solution = func(*args, **kwargs)
+        if func in cached_results:
+            cached_results[func][args] = solution
+        else:
+            cached_results[func] = {args: solution}
+        print("Calculating new result")
+        return solution
+
+    return inner
```

Comment:
In your dict `cached_results`   args should be a key, and the result of the function should be a value for this key. You don't need to make a func as a key, because you decorate this function and results will be available only for it.

Reply:
Thank you, it's clear for me now.